test/functional/mining_basic.py

```diff
@@ -53,7 +56,16 @@ def mine_chain(self):
         assert_equal(mining_info['blocks'], 200)
         assert_equal(mining_info['currentblocktx'], 0)
         assert_equal(mining_info['currentblocksize'], 1000)
+
+        self.log.info('test blockversion')
+        self.restart_node(0, extra_args=['-mocktime={}'.format(t), '-blockversion=1337'])
+        self.connect_nodes(0, 1)
+        assert_equal(1337, self.nodes[0].getblocktemplate()['version'])
+        self.restart_node(0, extra_args=['-mocktime={}'.format(t)])
+        self.connect_nodes(0, 1)
+        assert_equal(VERSIONBITS_TOP_BITS + (1 << VERSIONBITS_DEPLOYMENT_TESTDUMMY_BIT), self.nodes[0].getblocktemplate()['version'])
         self.restart_node(0)
+        # TODO: replace with connect_nodes_bi
```

Comment:
20167:
missing changes at line 94 https://github.com/dashpay/dash/pull/5842/commits/8844b66f40831794987c579042b7e32a7c5ebbca#diff-54aed8126fc19e55100de2f73c7ea9cac88ade36ead20734cf4298df043eff04R94

```
tmpl = node.getblocktemplate()
tmpl = node.getblocktemplate(NORMAL_GBT_REQUEST_PARAMS)
```

Reply:
[`NORMAL_GBT_REQUEST_PARAMS`](https://github.com/bitcoin/bitcoin/blob/fa9b48549ca39b862a10bcfd90e3eac2a0e8ad2e/test/functional/test_framework/blocktools.py#L60) is SegWit-specific and is not applicable for Dash.

Reply:
GBT means GenerateBlockTemplate

Reply:
I `git blame`'d the origin of the `NORMAL_GBT_REQUEST_PARAMS` variable ([bitcoin#19401](https://github.com/bitcoin/bitcoin/pull/19401)) and it's not an intrinsically SegWit-related PR (even if the value is). It has been backported and bitcoin#20167 has been adjusted accordingly.